packages/dara-core/js/auth/auth-wrapper.tsx

```diff
@@ -48,38 +86,61 @@ function DynamicAuthComponent(props: { component: AuthComponent }): JSX.Element
             .catch((err) => {
                 throw new Error(`Failed to import module ${props.component.py_module}`, err);
             });
-    }, [props.component]);
+    }, [props.component, importers]);
 
     return component;
 }
 
 /**
- * Simple Auth Wrapper component that makes the session request to the backend and then provides the session token down
- * to the rest of the app via context
+ * Simple Auth Wrapper component that initializes the token from localStorage,
+ * synchronizes token with localStorage and refreshes the token if it's about to expire.
+ * Renders the login/logout components and the children wrapped in PrivateRoute.
  *
  * @param props - the component props
  */
 function AuthWrapper(props: AuthWrapperProps): JSX.Element {
-    const [token, setToken] = useState<string>(() => getToken());
-    const { data: authComponents, isLoading } = useAuthComponents();
-
-    /**
-     * Set token handler - updates the token in state and local storage
-     *
-     * @param newToken new token
-     */
-    function onSetToken(newToken: string): void {
-        const key = getTokenKey();
-
-        if (newToken) {
-            localStorage.setItem(key, newToken);
-        } else {
-            localStorage.removeItem(key);
-        }
+    const { data: authConfig, isLoading } = useAuthConfig();
+    const isMounted = useRef(false);
 
-        setToken(newToken);
+    // set initial token from local storage as soon as we render
+    if (!isMounted.current) {
+        isMounted.current = true;
+        globalStore.setValue('sessionToken', getToken());
     }
 
+    // check token expiry every minute
+    useEffect(() => {
```

Comment:
What was the reasoning behind this interval? Why not just wait for a request to fail and then refresh the token?

Reply:
To be fair I can't quite recall, I think the idea originated when we were targeting a longer refresh time so we were trying to refresh ahead of time. With the low token time now we can get away without it.